File: app/src/main/java/br/com/roadmaps/mylibspinnersearch_2/MainActivity.java
package br.com.roadmaps.mylibspinnersearch_2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import java.util.ArrayList;
import java.util.List;

import br.com.roadmaps.mylibspinnersearch_2.listener.OnItemClickSpinnerSearch;
import br.com.roadmaps.mylibspinnersearch_2.test.AdapterTest;

public class MainActivity extends AppCompatActivity implements OnItemClickSpinnerSearch {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        List<String> list = new ArrayList<>();
        for (int i = 0; i < 20; i++){
            list.add("Item "+i);
        }
        SpinnerSearch spinner = findViewById(R.id.spinnerSearch2);
        spinner.setPopulateRecycleViewANDSetAdapter(new AdapterTest(list, this));
        spinner.setOnItemClickSpinner(this);
    }



    @Override
    public void onItemClickSpinner(String iten, int position) {
        Log.i("ITEM", "ACTIVITY_NEW"+iten +"  "+ position);
    }
}

File: sample_code/MainActivity.kt
package br.com.roadmaps.testespinnersearch2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.github.rodlibs.mylibspinnersearch_2.SpinnerSearch
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val list = arrayListOf<String>()
        for (i in 1..20) {
            list.add("item $i")
        }
        val spinner = findViewById<SpinnerSearch>(R.id.spinner)
        spinner.setPopulateRecycleView(list)
        spinner.setPopulateRecycleViewANDSetAdapter(AdapterCustomized(list))
        spinner.setOnItemClickSpinner { item, _ ->
            Log.i("SPINNER", item)
        }
    }
}
